Backpack/src/test/java/net/skyscanner/backpack/util/TestStateMachine.kt

```diff
@@ -46,14 +48,16 @@ class TestStateMachineResultScope<State>(
   val state: State,
 )
 
+@OptIn(ExperimentalCoroutinesApi::class)
 fun <SM : StateMachine<State, *>, State> testStateMachine(
   creator: CoroutineScope.(CoroutineDispatcher) -> SM,
   block: suspend TestStateMachineScope<SM, State>.() -> TestStateMachineResult
 ) {
-  val coroutineDispatcher = TestCoroutineDispatcher()
-  val coroutineScope = TestCoroutineScope() + coroutineDispatcher
+  val testCoroutineScheduler = TestCoroutineScheduler()
```

Comment:
we need to run everything on the same scheduler otherwise tests will fail now. also replaced the other bits with a non-deprecated version